lib/compat/wordpress-6.0/blocks.php

```diff
@@ -176,7 +176,8 @@ function build_comment_query_vars_from_block( $block ) {
 				} elseif ( 'oldest' === $default_page ) {
 					$comment_args['paged'] = 1;
 				} elseif ( 'newest' === $default_page ) {
-					$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+					$max_num_pages         = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+					$comment_args['paged'] = 0 === $max_num_pages ? 1 : $max_num_pages;
```

Comment:
Resyncing for the changes made when this code was committed into Core via changeset https://core.trac.wordpress.org/changeset/53336:

```suggestion
					$max_num_pages  = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
					if ( 0 !== $max_num_pages ) {
						$comment_args['paged'] = $max_num_pages;
					}
```

Why this change?

In `WP_Comment_Query` natively sets the default query args when they are not present. If there are no comments, then `'paged'` should not be set elsewhere, letting Core natively handle setting the defaults in one place; else, set the `'paged'` query arg with the `max_num_pages` value here.

Centralizing default states helps with maintenance as then there's only one place where the default is managed.

Also see @dream-encode's comment here https://github.com/WordPress/wordpress-develop/pull/2663#issuecomment-1116129178 (noting that he's one of the Comments component maintainers).